src/vbk/merkle.cpp

```diff
@@ -87,6 +87,10 @@ KeystoneArray getKeystoneHashesForTheNextBlock(const CBlockIndex* pindexPrev)
 
 uint256 TopLevelMerkleRoot(const CBlockIndex* prevIndex, const CBlock& block, bool* mutated)
 {
+    if (prevIndex == nullptr || !Params().isPopActive(prevIndex->nHeight + 1)) {
```

Comment:
what if POP is enabled from block 0?
Then, prevIndex is nullptr and merkle root will be calculated with regular BTC rules, but should be calculated with POP rules

Reply:
Code is taken from the Alex write-up. Should update it as well.